File: Mario.py
##Mario (Python Ver)
#Mahad Adeel, 06/09/20

try:
    while True:
        height = int(input("The height of the pyramid must be a full non-zero number less than 23\nIf a decimal is entered it will be rounded down\nEnter height: "))
        height += 0
        if not 0 < height <= 23:
            print("Enter a number within range")
        else:
            break
except:
    print("Enter a valid height.")

for i in range(1,height+1):
    print(((" "*(height-i))+("#"*i))+"#")

File: Mario.c
# include <stdio.h>

int height;
int i;
int x;

int main(){
   while (!(0 < height && height <= 23)) {
      printf("Enter height: ");
      while (scanf("%d", &height) == 0) {
         printf("Invalid input. Try again\nEnter height: ");
         int c;
         while((c=getchar())!='\n' && c!=EOF);
      }
      if (!(0 < height && height <= 23)) {
         printf("Enter a number within the range of 1-23\n");
      }
   }

   for (i = 1; i < (height + 1); ++i) {
      for (x = 0; x < (height - i); ++x) {
         printf(" ");
      }
      for (x = 1; x < i+1; ++x) {
         printf("#");
      }
      printf("#\n");
   }

   return 0;
}
